partner_aging/customer.py

```diff
@@ -0,0 +1,288 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2014 Ursa Informative Systems (<www.ursainfosystems.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+
+from openerp.osv import fields, osv
+from openerp import tools
+
+
+class account_aging_customer(osv.osv):
+    _name = 'partner.aging.customer'
+    _auto = False
+
+    def open_document(self, cr, uid, ids, context=None):
+        """
+        @description  Open document (invoice or payment) related to the
+                      unapplied payment or outstanding balance on this line
+        """
+        if not context:
+            context = {}
+        models = self.pool.get('ir.model.data')
+        # Get this line's invoice id
+        inv_id = self.browse(cr, uid, ids[0]).invoice_id.id
+
+        # if this is an unapplied payment(all unapplied payments hard-coded to
+        # -999), get the referenced voucher
+        if inv_id == -999:
+            ref = self.browse(cr, uid, ids[0]).invoice_ref
+            voucher_pool = self.pool['account.voucher']
+            # Get referenced customer payment (invoice_ref field is actually a
+            # payment for these)
+            voucher_id = voucher_pool.search(
```

Comment:
You should also check if there's no returned result. You can do this with  `[:1]` slicing